### PR TITLE
Add FeatureVoter Abstraction

### PR DESCRIPTION
**Pull request by Sherlouk**
Relates to #33 but doesn't necessarily close it. This PR separates "FeatureVoter" (an abstraction) from "FeatureVoterCountAPI" (the default concrete implementation).

This is not a breaking change, and will continue to default to using the CountAPI solution but does now give clients the ability to implement their own solution (such as their own persistence API).

### CODE REVIEW

File: Sources/Roadmap/RoadmapConfiguration.swift [hunk 1]
```diff
@@ -36,7 +34,19 @@ public struct RoadmapConfiguration {
         }
 
         self.roadmapJSONURL = roadmapJSONURL
-        self.namespace = namespace
+        self.voter = FeatureVoterCountAPI(namespace: namespace)
```
Comment (AvdLee) [inline Sources/Roadmap/RoadmapConfiguration.swift:37]: Let's add a property `voter` to the initializer with this as the default fallback:

```swift
    public init(roadmapJSONURL: URL, namespace: String? = nil, style: RoadmapStyle = RoadmapTemplate.standard.style, shuffledOrder: Bool = false, voter🙂FeatureVoter? = nil) {
    // ...
    self.voter = voter ?? FeatureVoterCountAPI(namespace: namespace)
```

**Review by AvdLee — CHANGES_REQUESTED** [review]
Yesss, I love this! Let's actually make this usable by adjusting the initializer and we're good to merge. Thanks as well for writing the test!

File: Sources/Roadmap/RoadmapConfiguration.swift (hunk 1 above)
Comment (Sherlouk) [reply Sources/Roadmap/RoadmapConfiguration.swift:37]: Is this needed? The namespace parameter is used only by CountAPI and so the expectation is if you're using the default initialiser (or explicitly providing the namespace) then that is your intent.

A separate initialiser had all the same parameters, except replacing the optional namespace with a required voter parameter.

Both options are present!

Comment (AvdLee) [reply Sources/Roadmap/RoadmapConfiguration.swift:37]: Ah, I completely missed the initializer underneath this comment, all good in that case! 

Last point: would you mind writing a little instruction in the readme to finish this feature? That way, others can find the option more easily!

Comment (Sherlouk) [reply Sources/Roadmap/RoadmapConfiguration.swift:37]: Absolutely, I'll update the README.

**Comment by Sherlouk** [comment]
README updated: I've removed the FAQ as it is now encapsulated in the paragraph above and so felt unnecessary to duplicate information.

**Comment by jordibruin** [comment]
This is so nice @Sherlouk !!

**Comment by shaps80** [comment]
Yep I agree. Very nice. Solve the dep injection of a custom voting backend.

I actually also made some great progress with my suggestions. I'll make a Pr over the weekend. It's almost done.

It'd work well alongside this change as well.

**Review by AvdLee — APPROVED** [review]
Love the quality of this contribution, thanks a lot for helping us out here!

**Comment by AvdLee** [comment]
@Sherlouk once the merge conflict is solved, we can merge it in!

**Comment by Sherlouk** [comment]
@AvdLee - done 🎉

**Comment by AvdLee** [comment]
@Sherlouk merged, thanks a lot!!

**Pull request merged**